idl/paging_state.idl.hh

```diff
@@ -6,6 +6,19 @@ enum class read_repair_decision : uint8_t {
 };
 }
 
+enum class bound_weight : int8_t {
+    before_all_prefixed = -1,
+    equal = 0,
+    after_all_prefixed = 1,
+}
+
+enum class partition_region : uint8_t {
+    partition_start,
+    static_row,
+    clustered,
+    partition_end,
+};
```

Comment:
This is just to support a full `position_in_partition` to be stored in the paging state, rather than just a clustering key. `position_in_partition` is the standard way to represent a position inside a partition. We probably don't need the entire range of positions it can represent, but I think it is beneficial to use the standard type, instead of coming up a narrower ad-hoc one and add conversions on the edges.